tests/integration/conftest.py

```diff
@@ -38,13 +38,188 @@ def tags():
+        random.choice(string.ascii_lowercase + string.digits) for _ in range(8)
+    )
+
+
+@pytest.fixture(scope="session")
+def vnet2():
+    yield "idem-vnet2-" + "".join(
+        random.choice(string.ascii_lowercase + string.digits) for _ in range(8)
+    )
+
+
+@pytest.fixture(scope="session")
+def subnet():
+    yield "idem-subnet-" + "".join(
+        random.choice(string.ascii_lowercase + string.digits) for _ in range(8)
+    )
+
+
+@pytest.fixture(scope="session")
+def test_vnet():
+    yield "idem-test-vnet-" + "".join(
+        random.choice(string.ascii_lowercase + string.digits) for _ in range(8)
+    )
+
+
+@pytest.fixture(scope="session")
+def test_subnet():
+    yield "idem-test-subnet-" + "".join(
+        random.choice(string.ascii_lowercase + string.digits) for _ in range(8)
+    )
+
+
+@pytest.fixture(scope="session")
+def public_ip_addr():
+    yield "idem-public-ip-" + "".join(
+        random.choice(string.ascii_lowercase + string.digits) for _ in range(8)
+    )
+
+
+@pytest.fixture(scope="session")
+def test_public_ip_addr():
+    yield "idem-test-public-ip-" + "".join(
+        random.choice(string.ascii_lowercase + string.digits) for _ in range(8)
+    )
+
+
+@pytest.fixture(scope="session")
+def route_table():
+    yield "idem-rt-table-" + "".join(
+        random.choice(string.ascii_lowercase + string.digits) for _ in range(8)
+    )
+
+
+@pytest.fixture(scope="session")
+def route():
+    yield "idem-rt-" + "".join(
+        random.choice(string.ascii_lowercase + string.digits) for _ in range(8)
+    )
+
+
+@pytest.fixture(scope="session")
+def load_balancer():
+    yield "idem-load-bal-" + "".join(
+        random.choice(string.ascii_lowercase + string.digits) for _ in range(8)
+    )
+
+
+@pytest.fixture(scope="session")
+def nsg():
+    yield "idem-nsg-" + "".join(
+        random.choice(string.ascii_lowercase + string.digits) for _ in range(8)
+    )
+
+
+@pytest.fixture(scope="session")
+def zone():
+    yield "idem.zone." + "".join(
+        random.choice(string.ascii_lowercase + string.digits) for _ in range(8)
+    )
+
+
+@pytest.fixture(scope="session")
+def availability_set():
+    yield "idem-avail-set-" + "".join(
+        random.choice(string.ascii_lowercase + string.digits) for _ in range(8)
+    )
+
+
+@pytest.fixture(scope="session")
+def network_interface():
+    yield "idem-net-iface-" + "".join(
+        random.choice(string.ascii_lowercase + string.digits) for _ in range(8)
+    )
+
+
+@pytest.fixture(scope="session")
+def test_network_interface():
+    yield "idem-test-net-iface-" + "".join(
+        random.choice(string.ascii_lowercase + string.digits) for _ in range(8)
+    )
+
+
+@pytest.fixture(scope="session")
+def local_network_gateway():
+    yield "idem-local-net-gateway-" + "".join(
+        random.choice(string.ascii_lowercase + string.digits) for _ in range(8)
+    )
+
+
+@pytest.fixture(scope="session")
+def ip_config():
+    yield "idem-ip-config-" + "".join(
```

Comment:
this is a lot of session fixtures... I think the only reason to have them is sharing names between different test *modules*. A `test-present` and `test_absent` function in the same module can just hard code the names (like I did with the secret tests) or use a module fixture, but the secret module needed to know the keyvault that was created in a separate module so I used a fixture for that. Are we sharing all of these names among multiple tests?